### PR TITLE
Using new BAG API for woonplaatsen detail page

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: src/api/bag/v1/woonplaatsen/detail-config.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Single as Woonplaatsen } from './types'
```
Comment (jonkoops) [inline src/api/bag/v1/woonplaatsen/detail-config.ts:1]: I would not put this file in the API directory, this is a change that specifically for the detail panels so let's keep the code there. The API directory is more for direct mapping of API methods and types, without additional transformation.

File: src/app/pages/MapPage/detail/DetailLinkList.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ const DetailLinkList: FunctionComponent<DetailLinkListProps> = ({ item, ...other
         if ('url' in link) {
           return (
             <StyledLink
-              key={link.url}
+              key={link.id ?? link.url}
```
Comment (jonkoops) [inline src/app/pages/MapPage/detail/DetailLinkList.tsx:38]: Why would the URL not be unique?

File: src/app/pages/MapPage/detail/getListFromApi.ts [hunk 3]
```diff
@@ -45,12 +45,14 @@ const getListFromApi = (
   }
 
   const results = normalize
-    ? await normalize(response.results || [response])
+    ? // @ts-ignore
+      // eslint-disable-next-line no-underscore-dangle
+      await normalize(response._embedded || response.results || [response])
```
Comment (jonkoops) [inline src/app/pages/MapPage/detail/getListFromApi.ts:50]: Can we add some tests for this?

File: src/map/sagas/nearest-details/nearest-details.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ export function* fetchNearestDetails(action) {
     const { uri } = yield call(fetchNearestDetail, location, layers, zoom)
     const isEmbed = yield select(isEmbedded)
 
-    if (uri) {
+    if (uri.type) {
```
Comment (jonkoops) [inline src/map/sagas/nearest-details/nearest-details.js:15]: Why `.type` here?

File: src/map/sagas/nearest-details/nearest-details.test.js [hunk 5]
```diff
@@ -36,7 +36,7 @@ describe('fetchNearestDetails', () => {
     },
   }
 
-  it('should call fetchNearestDetails and dispatch the correct actions if uri is returned', () => {
+  it.only('should call fetchNearestDetails and dispatch the correct actions if uri is returned', () => {
```
Comment (jonkoops) [inline src/map/sagas/nearest-details/nearest-details.test.js:39]: This change can be removed.

File: src/map/services/map-detail/map-detail.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import { endpointTypes } from '../map-services.config'
 import environment from '../../../environment'
+import { path as woonplaatsenPath } from '../../../api/bag/v1/woonplaatsen'
```
Comment (jonkoops) [inline src/map/services/map-detail/map-detail.ts:3]: Neat way of re-using this info :+1:

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/map/sagas/nearest-details/nearest-details.js (hunk 4 above)
Comment (chrisvanmook) [reply src/map/sagas/nearest-details/nearest-details.js:15]: we're now always returning an object with values that could also be null from `getDetailPageData` in `actions.js`. We could also return `null` but then we'll have to refactor a lot, so for now I left it like this.

File: src/app/pages/MapPage/detail/DetailLinkList.tsx (hunk 2 above)
Comment (chrisvanmook) [reply src/app/pages/MapPage/detail/DetailLinkList.tsx:38]: results might show twice, not sure why but it's not unique...

**Review by jonkoops — APPROVED** [review]

**Pull request merged**